pkg/util/limitrange/limitrange_test.go

```diff
@@ -219,6 +230,17 @@ func TestTotalRequest(t *testing.T) {
 				"example.com/gpu":     resource.MustParse("2"),
 			},
 		},
+		"include sidecar container": {
+			podSpec: &corev1.PodSpec{
+				InitContainers: containers[2:],
+				Containers:     containers[:2],
```

Comment:
Could we inline these lists for readability? If it is not too much additional effort, please refactor the other tests too

maybe some api like:
`CreateContainer().WithCpuReqs("1000m").WithMemoryReqs("2Gi").Sidecar()`

or even more simply just a few functions which create a few different pods, that you could mix and match in the various tests